reactos/drivers/usb/usbhub_new/pnp.c

```diff
@@ -1477,29 +1460,31 @@ USBH_FdoRemoveDevice(IN PUSBHUB_FDO_EXTENSION HubExtension,
 
     DPRINT("USBH_FdoRemoveDevice: HubExtension - %p\n", HubExtension);
 
-    PortData = HubExtension->PortData;
     HubDescriptor = HubExtension->HubDescriptor;
 
-    if (PortData && HubDescriptor)
+    if (HubDescriptor)
     {
         NumPorts = HubDescriptor->bNumberOfPorts;
 
         for (ix = 0; ix < NumPorts; ++ix)
         {
-            PortDevice = PortData->DeviceObject;
+            PortData = HubExtension->PortData + ix;
 
-            if (PortDevice)
+            if (PortData)
```

Comment:
pnp.c:1473: If there's a chance `HubExtension->PortData` can be NULL, you should keep the check at the top where you had it before. If you check it here, it will only be NULL for ix=0, it'll be a pointer like 0x00000020 for the other indexes.